### PR TITLE
kafka(ticdc): claim check do not use filepath join to build the full file path.

### LINKED ISSUE #10036: claim check file full path is not correct since duplicate slash is clean up
`filepath.join` is used to build the file full path, it makes the `s3://` to `s3:/`, this is not expected.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #10036

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: pkg/sink/kafka/claimcheck/claim_check.go [hunk 1]
```diff
@@ -102,7 +101,7 @@ func (c *ClaimCheck) WriteMessage(ctx context.Context, key, value []byte, fileNa
 
 // FileNameWithPrefix returns the file name with prefix, the full path.
 func (c *ClaimCheck) FileNameWithPrefix(fileName string) string {
-	return filepath.Join(c.storage.URI(), fileName)
+	return c.storage.URI() + "/" + fileName
```
Comment (CharlesCheung96) [inline pkg/sink/kafka/claimcheck/claim_check.go:104]: Please add unit test to cover this function.

Comment (nongfushanquan) [reply pkg/sink/kafka/claimcheck/claim_check.go:104]: if the configure end with "/", whether it is still work?

**Review by hicqu — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**